tests/v2/codecoverage/setuprancher/main.go

```diff
@@ -0,0 +1,511 @@
+func main() {
+	rancherConfig := new(rancher.Config)
+	config.LoadConfig(rancher.ConfigurationFileKey, rancherConfig)
+
+	kubeconfig, err := getRancherKubeconfig()
+	if err != nil {
+		logrus.Fatalf("error with getting kube config using corral: %v", err)
+	}
+
+	password, err := corral.GetCorralEnvVar(corralName, "bootstrap_password")
+	if err != nil {
+		logrus.Fatalf("error getting password %v", err)
+	}
+
+	// update deployment
+	err = updateRancherDeployment(kubeconfig)
+	if err != nil {
+		logrus.Fatalf("error updating rancher deployment: %v", err)
+	}
+
+	token, err := createAdminToken(password, rancherConfig)
+	if err != nil {
+		logrus.Fatalf("error with generating admin token: %v", err)
+	}
+
+	rancherConfig.AdminToken = token
+	config.UpdateConfig(rancher.ConfigurationFileKey, rancherConfig)
+	//provision clusters for test
+	session := session.NewSession()
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+	kubernetesVersions := clustersConfig.RKE1KubernetesVersions
+	cnis := clustersConfig.CNIs
+	nodesAndRoles := clustersConfig.NodesAndRolesRKE1
+
+	client, err := rancher.NewClient("", session)
+	if err != nil {
+		logrus.Fatalf("error creating admin client: %v", err)
+	}
+
+	err = postRancherInstall(client)
+	if err != nil {
+		logrus.Fatalf("error with admin user service account secret: %v", err)
+	}
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = passwordgenerator.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	if err != nil {
+		logrus.Fatalf("error creating admin client: %v", err)
+	}
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	if err != nil {
+		logrus.Fatalf("error creating standard user client: %v", err)
+	}
+
+	// create admin cluster
+	adminClusterNames, err := createTestCluster(client, client, 1, "admintestcluster", cnis[0], kubernetesVersions[0], nodesAndRoles)
+	if err != nil {
+		logrus.Fatalf("error creating admin user cluster: %v", err)
+	}
+
+	// create standard user clusters
+	standardClusterNames, err := createTestCluster(standardUserClient, client, 2, "standardtestcluster", cnis[0], kubernetesVersions[0], nodesAndRoles)
+	if err != nil {
+		logrus.Fatalf("error creating standard user clusters: %v", err)
+	}
+
+	//update userconfig
+	userClusterConfig := UserClusterConfig{}
+	userClusterConfig.Token = standardUserClient.Steve.Opts.TokenKey
+	userClusterConfig.Username = newUser.Name
+	userClusterConfig.AdminPassword = adminPassword
+	userClusterConfig.Clusters = standardClusterNames
+
+	rancherConfig.ClusterName = adminClusterNames[0]
+	config.UpdateConfig(rancher.ConfigurationFileKey, rancherConfig)
+
+	err = writeToConfigFile(userClusterConfig)
```

Comment:
Why create a new config file if we can update the existing `CATTLE_CONFIG_TEST`?

Reply:
This is a separate config file, this config file is used to glue python and golang test runs. The python tests don't use CATTLE_CONFIG_TEST

Reply:
Python tests doesn't use `CATTLE_CONFIG_TEST` but neither the `userClusterConfig` which in this case is used only to transport data for configuring the Python framework.

The same can't be done just updating `CATTLE_CONFIG_TEST` and use `yq` as done in the jenkinsfile over `userclusterconfig.yaml` ?

I am just curious, if this is just to separate the configuration for clarity im good.

Reply:
Simply because we need the distinction between the admin clusters and the standard user clusters.

Reply:
I see I thought we could create an additional configuration key in the `CATTLE_CONFIG_TEST` to differentiate, but I guess it's clearer/easier this way.

Reply:
We don't one of the main designs of the framework is dynamic generate a user token rather than having to manually create

Reply:
Ok just to wrap up. Let me know if this is correct.

Like what happens with `rancher.adminToken` in `CATTLE_CONFIG_TEST` where `createAdminToken` is used in go and then assigned to the struct `rancherConfig.AdminToken = token`  to then being applied using `config.UpdateConfig` back to `CATTLE_CONFIG_TEST`

But is not possible to have the same with a new configuration key using a struct like `userClusterConfig`

Reply:
It definitely is possible to have a new configuration key in `CATTLE_CONFIG_TEST` to use something like `userClusterConfig`, the reason I didn't do that was because I felt like code coverage setup was very different from the rest of our testing, and felt like it didn't belong in the `CATTLE_CONFIG_TEST` especially since the main/only reason I have it is to setup our python tests both integration and validation.